### PR TITLE
Changes to become v1.x

### PR DESCRIPTION
**Pull request by aredridel**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -15,51 +15,198 @@
...
+        permutron.raw(search, function (candidate, next) {
+            var resolved = path.resolve.apply(null, candidate);
+            limitStat(resolved, function (err, stat) {
+                if (!err && stat.isFile()) {
+                    debug('found "%s"', resolved);
+                    cb(null, resolved);
+                } else if ((!err && stat.isDirectory()) || (err && err.code === 'ENOENT')) {
+                    next();
+                } else {
+                    cb(err);
```
Comment (grawk) [inline index.js:94]: I guess it looks like there is no need to check that err is defined here..

Comment (aredridel) [inline index.js:94]: Yeah. I'm not super happy with that logic -- it works, it's correct, but it's not obvious.

File: index.js [hunk 2]
```diff
@@ -15,51 +15,198 @@
...
-    if (configs['view engine'] === 'dust') {
-        munger.wrapDustOnLoad('dust', configs, settings.cache);
+    /**
+     * Render with the given `options` and callback `fn(err, str)`.
+     *
+     * @param {Object} options
+     * @param {Function} fn
+     * @api private
+     */
+    proto.render = function render(options, fn) {
```
Comment (grawk) [inline index.js:116]: why `fn` instead of `cb`?

Comment (aredridel) [inline index.js:116]: Ported from my patch to Express. The term it used was `fn`.

Comment (grawk) [inline index.js:116]: It does introduce an internal inconsistency to this module. Just thought that was worth noting. I hope to be to a place to provide better feedback than this soon. Just in my first read-through.

Comment (aredridel) [inline index.js:116]: Yep. It's not a justification, just an explanation.

Comment (grawk) [inline index.js:116]: Regarding the express patch, I don't see that mentioned in any of the commit messages for this PR. How is the association between this code change and the express patch noted in this software?

Comment (aredridel) [inline index.js:116]: Heh, it's not! https://github.com/strongloop/express/pull/2653 is the pull for express 5

Comment (aredridel) [inline index.js:116]: Added a link to the README

File: README.md [hunk 3]
```diff
@@ -1,88 +1,93 @@
...
-$ npm test
 
-To run coverage
-$ npm run-script cover
+To run coverage:
 
-To run lint
-$ npm run-script lint
+```shell
+npm run cover
```
Comment (grawk) [inline README.md:92]: I had to run `npm run-script cover`. Is this due to my npm version? (2.5.1)

Comment (aredridel) [inline README.md:92]: No, run has always been an alias for run-script. What happened when you did `npm run cover`?

File: test/fixtures/compile-test-data.sh [hunk 4]
```diff
@@ -0,0 +1,19 @@
+#!/bin/bash
```
Comment (grawk) [inline test/fixtures/compile-test-data.sh:1]: Was this script used once to compile test data? It doesn't seem to be run by any other module or command. Also, when running it locally, it complains `localizr: command not found`

File: README.md (hunk 3 above)
Comment (grawk) [inline README.md:92]: oh. sorry. i was simply using `npm cover` and then straight to `npm run-script cover`.. I didn't properly read the doc above and try `npm run`. That works for me now that I've tried it.

Comment (grawk) [inline README.md:92]: Awesome coverage metrics btw!!

File: test/fixtures/compile-test-data.sh (hunk 4 above)
Comment (aredridel) [inline test/fixtures/compile-test-data.sh:1]: It is, and I should probably make localizr a devDependency because of that.

Comment (aredridel) [inline test/fixtures/compile-test-data.sh:1]: (the test .js files were woefully out of date, and dust doesn't guarantee compatibility if you use a different compiler and runtime version)

Comment (grawk) [inline test/fixtures/compile-test-data.sh:1]: When you say "it is" do you mean it is run by another command somewhere and I just didn't find it?

Comment (aredridel) [inline test/fixtures/compile-test-data.sh:1]: Ah, no, it is used for that. It's run entirely manually. Perhaps I need to expose it in package.json ;-)

Comment (grawk) [inline test/fixtures/compile-test-data.sh:1]: Exposing it in package.json could be good. Or a couple comment lines within this file. Or markdown documentation in this directory.

Comment (aredridel) [inline test/fixtures/compile-test-data.sh:1]: Exposed.

File: .gitignore [hunk 5]
```diff
@@ -2,3 +2,6 @@
 node_modules/
 coverage/
 npm-debug.log
+*.swp
+test/fixtures/tmp/
+nyc_output/
```
Comment (grawk) [inline .gitignore]: I think this needs to be `.nyc_output/`

Comment (aredridel) [inline .gitignore]: Fixed

**Comment by grawk** [comment]
Is there a sample express app anywhere currently using the v1.x pre-release version of engine-munger?

**Comment by aredridel** [comment]
Working on integrating it into a kraken app right now.

**Comment by aredridel** [comment]
Also, the tests fire up a complete app.

**Comment by grawk** [comment]
Ah right. I guess the root of my question was really "which versions of adaro and other things go along with this?" but it's really right there in package.json already! Aside from the picky stuff I already noted I don't have any other concerns. But let me digest this some more to make sure I can learn as much as possible before giving a thumbs up.

**Comment by aredridel** [comment]
awesome.  It's really the design and redrawing of module boundaries I want to make sure make sense.

**Comment by grawk** [comment]
In terms of structure, what do you think about externalizing the `proto` patch from the exported  `makeViewClass` function? I'm just thinking in terms of the fact that the patch will hopefully go away when the PR is accepted to express and this module is released to match `express@5.x`.

_edit_: Let me restate this: would there be sufficient value to include the express view patch verbatim in this module, and then externalize all the "non-patch" stuff (like bcp47/i18n resolution)? That could make more clear how we will eventually utilize express once 5.x is released.

File: test/index.js [hunk 6]
```diff
@@ -1,20 +1,14 @@
 'use strict';
-var dustjs = require('dustjs-linkedin'),
-    test = require('tape'),
-    engineMunger = require('../index'),
-    testData = require('./fixtures/testConfig'),
-    freshy = require('freshy');
-
+var test = require('tap').test;
+var freshy = require('freshy');
```
Comment (grawk) [inline test/index.js]: freshy isn't used in this module

Comment (aredridel) [inline test/index.js]: Good catch

**Comment by aredridel** [comment]
Mostly the prototype creation is inside `makeViewClass` because the `conf` variable is shared with the internals but private.

None of that is performance critical, since a new view class is created once per app in normal use, so having it be properly private by scoping made me happy.

**Comment by grawk** [comment]
Ok. Officially in love with this PR. :+1:

**Comment by pvenkatakrishnan** [comment]
:+1:

**Pull request merged**